packages/ckeditor5-image/src/image/utils.js

```diff
@@ -133,6 +147,58 @@ export function getViewImgFromWidget( figureView ) {
 	return figureChildren.find( viewChild => viewChild.is( 'element', 'img' ) );
 }
 
+// Creates a view element representing the image.
+//
+//		<figure class="image"><img></img></figure>
+//
+// Note that `alt` and `src` attributes are converted separately, so they are not included.
+//
+// @private
+// @param {module:engine/view/downcastwriter~DowncastWriter} writer
+// @param {'image'|'imageInline'} imageType The type of created image.
+// @returns {module:engine/view/containerelement~ContainerElement}
+export function createImageViewElement( writer, imageType ) {
+	const emptyElement = writer.createEmptyElement( 'img' );
+
+	const container = imageType === 'image' ?
+		writer.createContainerElement( 'figure', { class: 'image' } ) :
+		writer.createContainerElement( 'span', { class: 'image-inline' } );
+
+	writer.insert( writer.createPositionAt( container, 0 ), emptyElement );
+
+	return container;
+}
+
+// A function returning a {@link module:engine/view/matcher~Matcher} callback for a particular type of View images.
+//
+// @param {'image'|'imageInline'} matchImageType The type of created image.
+// @returns {Function}
+export function getImageTypeMatcher( matchImageType, editor ) {
+	if ( editor.plugins.has( 'ImageInline' ) ^ editor.plugins.has( 'ImageBlock' ) ) {
+		return {
+			name: 'img',
+			attributes: {
+				src: true
+			}
+		};
+	}
```

Comment:
I don't like using the binary xor operator for logical conditions. 

Also, this upcast conversion is always needed for block image (as mentioned in another comment).

Reply:
This upcast could work only when there is one plugin enabled. For both enabled there is another upcast returned below, which differentiates if `img` is block or inline depending on the context. About `xor` operator - without it, the logic will be more complex and this is the perfect situation for using it. Ofc. I could rewrite it, but first, please @oleq confirm that we don't want `xor` operators in the code :)